### DISCUSSION TITLE
Poor Performance WSL? (For the love of FlashAttention)

### DISCUSSION THREAD

**Discussion by nbollman** [oobabooga/text-generation-webui#4271] [Q&A] [answered]
Dual booting and WSL, for the love of FlashAttention...
Windows was disappointed by not being able to use FlashAttention so -~2GBvram, but still loads Mistral-7b ~7sec
Linux load Mistral 7b Llama 2 ~4sec and FlashAttention works so +2GBvram yay!
WSL (im inexperienced) everything seems to be working... including flash attention, but models load horrendously slow ~170secs for Mistral-7b using exllama2...
dunno what Im doing wrong, Ive started over too many times...

Here is setup overview
Win11 Latest Nvidia drivers RTX3090/Cuda 12.2>WSL/Ubuntu22.04 Cuda12.2 > Conda environment
Do I have to install an older version of cuda on WSL or something?

**Comment by nbollman** [comment, marked as answer]
Ok, Ive figured out my own probem... I have two m.2 drives 2TB 980 gen4 and a 4TB Crucial... everything trying to load from outside the WSL install on the main C: loads slow... WSL wants the directory structure within WSL check...

**Comment by Zorr0S** [comment]
Nice! imma try it, btw do you recommend using Cuda 12.2 or 11.8?
